linera-sdk/src/lib.rs

```diff
@@ -79,9 +79,23 @@ pub struct ViewStateStorage<A>(std::marker::PhantomData<A>);
 #[async_trait]
 pub trait Contract: Sized {
     /// Message reports for application execution errors.
-    type Error: Error;
+    type Error: Error + From<bcs::Error>;
     /// The desired storage backend to use to store the application's state.
     type Storage;
+    /// Initialization Arguments.
+    type InitializationArguments: DeserializeOwned + Send;
+    /// Execute Operation Arguments.
+    type Operation: DeserializeOwned + Send;
+    /// Application Call Arguments.
+    type ApplicationCallArguments: DeserializeOwned + Serialize + Send;
+    /// The Application Effect.
+    type Effect: DeserializeOwned + Serialize + Send + std::fmt::Debug;
+    /// A Session Call.
+    type SessionCall: DeserializeOwned + Send;
+    /// The response type of an application call.
+    type Response: Serialize + Send;
```

Comment:
Seems like this should be called `ApplicationCallReturnValue` then (and moved above).

Reply:
Oh I see. You wanted to re-use the same type as sessions. But I don't see why they would be the same type in general. Looks like we need `SessionCallReturnValue`.

Reply:
Well it's wrapping an `ApplicationCallResult` right? I assumed they were the same.